datadoc/backend/DataDocMetadata.py

```diff
@@ -73,14 +74,11 @@ def __init__(self, dataset):
 
         self.read_metadata_document()
 
-    def get_dataset_state(
-        self, dataset_path: pathlib.Path = None
-    ) -> Optional[DatasetState]:
+    def get_dataset_state(self, dataset: str) -> Optional[DatasetState]:
         """Use the path to attempt to guess the state of the dataset"""
-
-        if dataset_path is None:
-            dataset_path = self.dataset_full_path
-        dataset_path_parts = list(dataset_path.parts)
+        if dataset is None:
+            return None
+        dataset_path_parts = list(pathlib.Path(dataset).parts)
         if "kildedata" in dataset_path_parts:
```

Comment:
This seems very fragile. Isn't it better to force the users to select the correct dataset state?

Reply:
Yes, it is fragile, I believe it is done in the interests of _trying_ to make the users' job easier (i.e. minimise how much information they must input).

On the other hand, if the path doesn't include the dataset state the only consequence is the user enters it manually.

Reply:
Ok, but maybe the implementation should follow this naming standard: https://statistics-norway.atlassian.net/wiki/spaces/MPD/pages/3170140229/Navnestandard+og+versjonering+av+datasett+i+DAPLA#Navngiving-av-mapper-og-datasett-(filer)-i-DAPLAs-datalager

Which would mean that "inndata" should be more widely used than "kildedata".